File: FirmwareGO/application/services/interfaces/IFirmwareService.go
package interfaces

import (
	. "FirmwareGO/application/dtos"
	"context"
)

type IFirmwareService interface {
	CreateFileSend(ctx context.Context, newFileSendDto CreateFileSendDTO) (*FileSendResponseDTO, error)
	GetFirmwareHistoryByDeviceId(ctx context.Context, deviceId int) ([]*FileSendResponseDTO, error)
}

File: DeviceFirmwareOrchestrator/Application/ApplicationServices/Interfaces/IFirmwareService.cs
using Application.DTOs.Firmware;

namespace Application.ApplicationServices.Interfaces;

public interface IFirmwareService
{
    Task<FileSendResponseDTO> CreateFileSendAsync(CreateFileSendDTO newFileSendDto);
    Task<IEnumerable<FileSendResponseDTO>> GetFirmwareHistoryForDeviceAsync(int deviceId);
}